dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserCredentialsStore.java

```diff
@@ -73,4 +73,12 @@ public UserCredentials getUserCredentialsByLdapId( String ldapId )
         query.setParameter( "ldapId", ldapId );
         return query.uniqueResult();
     }
+
+    @Override
+    public UserCredentials getUserCredentialsBySecret( String secret )
```

Comment:
Let's introduce a new property for this we use only for invites/password reset. The secret field is used as a shared secret for 2FA and we should be very careful not expose it through the API at all.

Reply:
New property made, the name is "idtoken", I also removed an old property not in use anymore called "restorecode"